File: Find the Closest Element in BST[5-June-2023]/solution.cpp
//  >> C++ CODE <<

class Solution {
    public:
    int res[1];
    
    int minDiff(Node* root, int K) {
        res[0] =1e9;
        solve(root, K);
        return res[0];
    }
    
    void solve(Node* root, int k) {
        if (root == nullptr) return;
        
        res[0] =min(res[0], abs(root->data - k));
        
        if (k < root->data) {
            solve(root->left, k);
        } else {
            solve(root->right, k);
        }
    }
};

File: Find the Closest Element in BST[5-June-2023]/Solution.java
//  >> JAVA CODE <<

class Solution {
    private static int[] res;
    
    int minDiff(Node root, int K) {
        res = new int[]{Integer.MAX_VALUE};
        solve(root, K);
        return res[0];
    }
    
    static void solve(Node root, int k) {
        if (root == null) return;
        
        res[0] = Math.min(res[0], Math.abs(root.data - k));
        
        if (k < root.data) {
            solve(root.left, k);
        } else {
            solve(root.right, k);
        }
    }
}